hoomd/write/hdf5.py

```diff
@@ -209,7 +209,12 @@ def _initialize_datasets(self, log_dict):
             chunk_size = None
             if category == "scalar":
                 data_shape = (1,)
-                dtype = "f8" if isinstance(value, float) else "i8"
+                if isinstance(value, (np.integer, np.floating, np.bool_)):
```

Comment:
```suggestion
                if isinstance(value, (np.number, np.bool_)):
```
This covers integer, floating, and complex types which h5py supports: https://docs.h5py.org/en/stable/faq.html#what-datatypes-are-supported